# KPN1053 alignments (2018-10-22)

## Overview
In this part the following steps were executed:
  - Download ~ 200 references, split them to chromosomes and plasmids
  - Select the best chromosome reference using RefRank tool
  - Align Klebsiella reads to selected chromosome reference
  - Analyze alignment quality
  - Try to find plasmids using PlasmidFinder against raw reads and against assembled Klebsiella genome

See all results here: http://195.208.116.225/bioinf/Klebsiella/9_alignment/

## Installation requirements
See https://github.com/dariader/Klebsiella/blob/alignments/README.md

## 1. Data preperation

You can find more instructions here: https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/

### 1.1 Find and download all available Klebsiella pneumaniae genomes

First of all, we want to download all available complete Klebsiella pneumaniae genomes.
This command will download NCBI assembly summary, which contains all available bacteria genomes:
```
wget ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt
```

Then we can find all Klebsiella pneumaniae complete genomes:
```
awk -F "\t" '$8=="Klebsiella pneumoniae" &&  $12=="Complete Genome" && $11=="latest" {print $20}' assembly_summary.txt > klebsiella_pneumaniae_genomes.txt
```

Let's calculate a number of found genomes:
```
wc -l klebsiella_pneumaniae_genomes.txt 

212 klebsiella_pneumaniae_genomes.txt
```

Construct full FTP paths to selected genomes:
```
awk 'BEGIN { FS=OFS="/"; filesuffix="genomic.fna.gz" }{ ftpdir=$0;asm=$10;file=asm"_"filesuffix; print ftpdir,file }' klebsiella_pneumaniae_genomes.txt > klebsiella_pneumaniae_genome_paths.txt
```

Create a directory for Klebsiella genomes:

```
 mkdir klebsiella_reference
 cd ./klebsiella_reference/
```

// TODO: download GFF files for best genome later
Download all selected genomes:
```
time cat ../klebsiella_pneumaniae_genome_paths.txt | parallel --gnu "wget {}"

real	45m28.196s
user	0m2.496s
sys	0m6.151s
```

Unzip genomes:
```
gunzip ../../data/klebsiella_reference/*.fna.gz
```

### 1.2 Split genomes to chromosomes and plasmids

Split genomes to per-sequence files using faidx tool:
```
time ls ../../../data/klebsiella_reference/ | xargs -n 1 basename | parallel --gnu "faidx -x ../../../data/klebsiella_reference/{}"
```

To figure out, which files are chromosomes and which are plasmids, we can use assembly report from NCBI.

Create a directory for assembly reports:
```
mkdir klebsiella_assembly_reports
```

Create full FTP paths to assembly reports:
```
awk 'BEGIN { FS=OFS="/"; filesuffix="assembly_report.txt" }{ ftpdir=$0;asm=$10;file=asm"_"filesuffix; print ftpdir,file }' klebsiella_pneumaniae_genomes.txt > klebsiella_pneumaniae_genome_reports.txt

wc -l klebsiella_pneumaniae_genome_reports.txt
212 klebsiella_pneumaniae_genome_reports.txt
```

Download all assembly reports:
```
time cat ../klebsiella_pneumaniae_genome_reports.txt | parallel --gnu "wget {}"

real	60m39.766s
user	0m2.153s
sys	0m3.175s
```

Select all chromosome names and plasmid names into separete files:
```
awk -F '\t' '{if ($4 =="Chromosome") print $7}' klebsiella_assembly_reports/GCF_* > klebsiella_chromosomes.txt
awk -F '\t' '{if ($4 =="Plasmid") print $7}' klebsiella_assembly_reports/GCF_* > klebsiella_plasmids.txt

wc -l klebsiella_chromosomes.txt
212 klebsiella_chromosomes.txt

wc -l klebsiella_plasmids.txt
728 klebsiella_plasmids.txt
```

Construct full paths to chromosome and plasmid file names:
```
awk 'BEGIN { fileprefix="/home/ubuntu/bioinf/Klebsiella/9_alignment/single_sequences" }{ chr=$1;file=fileprefix"/"chr".fna" ; print file }' ../../data/klebsiella_chromosomes.txt > klebsiella_chromosome_paths.txt

awk 'BEGIN { fileprefix="/home/ubuntu/bioinf/Klebsiella/9_alignment/single_sequences" }{ plsmd=$1;file=fileprefix"/"plsmd".fna" ; print file }' ../../data/klebsiella_plasmids.txt > klebsiella_plasmid_paths.txt
```

## 2. Select best chromosome reference
### 2.1 Run refRank against list of chromosome files

Run RefRank using chromosome sequences as target refernce files:
```
time ../../software/refRank/refrank.py --fastq ../2_kpn1053_trimmomatic/2/SRR6480154_1_P.fastq ../2_kpn1053_trimmomatic/2/SRR6480154_2_P.fastq --ref $(cat klebsiella_chromosome_paths.txt)
```

The best chromosome sequence is NZ_CP032222.1.fna.
See: https://www.ncbi.nlm.nih.gov/assembly/GCF_003571745.1/, https://www.ncbi.nlm.nih.gov/biosample/SAMN04014887 
Let's find a whole genome, which contains selected chromosome:
```
grep -l NZ_CP032222 ../../data/klebsiella_assembly_reports/*.txt

../../data/klebsiella_assembly_reports/GCF_003571745.1_ASM357174v1_assembly_report.txt
```

## 3. Align using Bowtie2

### 3.1 Index files
```
time bowtie2-build ~/bioinf/data/klebsiella_reference/GCF_003571745.1_ASM357174v1_genomic.fna NZ_CP032222

Settings:
  Output files: "NZ_CP032222.*.bt2"
  Line rate: 6 (line is 64 bytes)
  Lines per side: 1 (side is 64 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  /home/ubuntu/bioinf/data/klebsiella_reference/GCF_003571745.1_ASM357174v1_genomic.fna
Building a SMALL index
Reading reference sizes
  Time reading reference sizes: 00:00:00
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
bmax according to bmaxDivN setting: 1442179
Using parameters --bmax 1081635 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 1081635 --dcv 1024
Constructing suffix-array element generator
Building DifferenceCoverSample
  Building sPrime
  Building sPrimeOrder
  V-Sorting samples
  V-Sorting samples time: 00:00:00
  Allocating rank array
  Ranking v-sort output
  Ranking v-sort output time: 00:00:00
  Invoking Larsson-Sadakane on ranks
  Invoking Larsson-Sadakane on ranks time: 00:00:00
  Sanity-checking and returning
Building samples
Reserving space for 12 sample suffixes
Generating random suffixes
QSorting 12 sample offsets, eliminating duplicates
QSorting sample offsets, eliminating duplicates time: 00:00:00
Multikey QSorting 12 samples
  (Using difference cover)
  Multikey QSorting samples time: 00:00:00
Calculating bucket sizes
Splitting and merging
  Splitting and merging time: 00:00:00
Avg bucket size: 5.76872e+06 (target: 1081634)
Converting suffix-array elements to index image
Allocating ftab, absorbFtab
Entering Ebwt loop
Getting block 1 of 1
  No samples; assembling all-inclusive block
  Sorting block of length 5768716 for bucket 1
  (Using difference cover)
  Sorting block time: 00:00:01
Returning block of 5768717 for bucket 1
Exited Ebwt loop
fchr[A]: 0
fchr[C]: 1246607
fchr[G]: 2883405
fchr[T]: 4520178
fchr[$]: 5768716
Exiting Ebwt::buildToDisk()
Returning from initFromVector
Wrote 6118003 bytes to primary EBWT file: NZ_CP032222.1.bt2
Wrote 1442184 bytes to secondary EBWT file: NZ_CP032222.2.bt2
Re-opening _in1 and _in2 as input streams
Returning from Ebwt constructor
Headers:
    len: 5768716
    bwtLen: 5768717
    sz: 1442179
    bwtSz: 1442180
    lineRate: 6
    offRate: 4
    offMask: 0xfffffff0
    ftabChars: 10
    eftabLen: 20
    eftabSz: 80
    ftabLen: 1048577
    ftabSz: 4194308
    offsLen: 360545
    offsSz: 1442180
    lineSz: 64
    sideSz: 64
    sideBwtSz: 48
    sideBwtLen: 192
    numSides: 30046
    numLines: 30046
    ebwtTotLen: 1922944
    ebwtTotSz: 1922944
    color: 0
    reverse: 0
Total time for call to driver() for forward index: 00:00:02
Reading reference sizes
  Time reading reference sizes: 00:00:00
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
  Time to reverse reference sequence: 00:00:00
bmax according to bmaxDivN setting: 1442179
Using parameters --bmax 1081635 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 1081635 --dcv 1024
Constructing suffix-array element generator
Building DifferenceCoverSample
  Building sPrime
  Building sPrimeOrder
  V-Sorting samples
  V-Sorting samples time: 00:00:00
  Allocating rank array
  Ranking v-sort output
  Ranking v-sort output time: 00:00:00
  Invoking Larsson-Sadakane on ranks
  Invoking Larsson-Sadakane on ranks time: 00:00:00
  Sanity-checking and returning
Building samples
Reserving space for 12 sample suffixes
Generating random suffixes
QSorting 12 sample offsets, eliminating duplicates
QSorting sample offsets, eliminating duplicates time: 00:00:00
Multikey QSorting 12 samples
  (Using difference cover)
  Multikey QSorting samples time: 00:00:00
Calculating bucket sizes
Splitting and merging
  Splitting and merging time: 00:00:00
Avg bucket size: 5.76872e+06 (target: 1081634)
Converting suffix-array elements to index image
Allocating ftab, absorbFtab
Entering Ebwt loop
Getting block 1 of 1
  No samples; assembling all-inclusive block
  Sorting block of length 5768716 for bucket 1
  (Using difference cover)
  Sorting block time: 00:00:01
Returning block of 5768717 for bucket 1
Exited Ebwt loop
fchr[A]: 0
fchr[C]: 1246607
fchr[G]: 2883405
fchr[T]: 4520178
fchr[$]: 5768716
Exiting Ebwt::buildToDisk()
Returning from initFromVector
Wrote 6118003 bytes to primary EBWT file: NZ_CP032222.rev.1.bt2
Wrote 1442184 bytes to secondary EBWT file: NZ_CP032222.rev.2.bt2
Re-opening _in1 and _in2 as input streams
Returning from Ebwt constructor
Headers:
    len: 5768716
    bwtLen: 5768717
    sz: 1442179
    bwtSz: 1442180
    lineRate: 6
    offRate: 4
    offMask: 0xfffffff0
    ftabChars: 10
    eftabLen: 20
    eftabSz: 80
    ftabLen: 1048577
    ftabSz: 4194308
    offsLen: 360545
    offsSz: 1442180
    lineSz: 64
    sideSz: 64
    sideBwtSz: 48
    sideBwtLen: 192
    numSides: 30046
    numLines: 30046
    ebwtTotLen: 1922944
    ebwtTotSz: 1922944
    color: 0
    reverse: 1
Total time for backward call to driver() for mirror index: 00:00:02

real	0m3.589s
user	0m3.250s
sys	0m0.340s
```

### 3.2 Align 
```
time bowtie2 -x NZ_CP032222 -q -1 ../2_kpn1053_trimmomatic/2/SRR6480154_1_P.fastq -2 ../2_kpn1053_trimmomatic/2/SRR6480154_2_P.fastq -S NZ_CP032222
777108 reads; of these:
  777108 (100.00%) were paired; of these:
    96939 (12.47%) aligned concordantly 0 times
    661918 (85.18%) aligned concordantly exactly 1 time
    18251 (2.35%) aligned concordantly >1 times
    ----
    96939 pairs aligned concordantly 0 times; of these:
      37803 (39.00%) aligned discordantly 1 time
    ----
    59136 pairs aligned 0 times concordantly or discordantly; of these:
      118272 mates make up the pairs; of these:
        114410 (96.73%) aligned 0 times
        1935 (1.64%) aligned exactly 1 time
        1927 (1.63%) aligned >1 times
92.64% overall alignment rate

real	5m30.753s
user	5m24.806s
sys	0m4.386s
```

## 4. Variant calling
