# Check reference for all samples

## Overview
In this part the following steps were executed:
  - 

See all results here: http://195.208.116.225/bioinf/Klebsiella/9_alignment/

## Installation requirements
See https://github.com/dariader/Klebsiella/blob/alignments/README.md

## 1. Preperation

### 1. Run Trimmomatic

```
$ python3 ~/bioinf/software/my_scripts/run_trimmomatic.py -file ~/bioinf/data/PRJNA429779/ -params "-phred33 ILLUMINACLIP:/home/ubuntu/bioinf/software/Trimmomatic-0.38/adapters/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:4:20 MINLEN:36"
...
```

### 1.2 Run FastqC to check results

```
$ time fastqc -o ./fastqc2 -f fastq ./trimmomatic/*_p.fastq
...
real	20m42.735s
user	21m0.000s
sys	0m32.840s
```

## 2. Run Refrank

### 2.1

``
$ python3 ~/bioinf/software/my_scripts/run_refrank.py -dir ~/bioinf/Klebsiella/10_other_samples/trimmomatic
...
117551.93user 7924.95system 10:17:46elapsed 338%CPU (0avgtext+0avgdata 525208maxresident)k
2263112inputs+689274536outputs (63major+1079999605minor)pagefaults 0swaps
``

## 3. Alignments

### 3.1. Multiple alignments with MUSCLE

Install MUSCLE: http://www.drive5.com/muscle/downloads.htm

### 3.2 Bowtie2 alignment

```
$ time bowtie2-build ~/bioinf/data/klebsiella_reference/GCF_003571745.1_ASM357174v1_genomic.fna NZ_CP032222
```

```
$ python3 ~/bioinf/software/my_scripts/run_bowtie2.py -dir ~/bioinf/Klebsiella/10_other_samples/trimmomatic
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480150_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480150_2_p.fastq -S SRR6480150.sam > SRR6480150_1_p_fastq_SRR6480150_1_p_fastq__bowtie2_log
663844 reads; of these:
  663844 (100.00%) were paired; of these:
    209167 (31.51%) aligned concordantly 0 times
    439298 (66.17%) aligned concordantly exactly 1 time
    15379 (2.32%) aligned concordantly >1 times
    ----
    209167 pairs aligned concordantly 0 times; of these:
      100548 (48.07%) aligned discordantly 1 time
    ----
    108619 pairs aligned 0 times concordantly or discordantly; of these:
      217238 mates make up the pairs; of these:
        204220 (94.01%) aligned 0 times
        6040 (2.78%) aligned exactly 1 time
        6978 (3.21%) aligned >1 times
84.62% overall alignment rate
322.38user 4.16system 5:26.22elapsed 100%CPU (0avgtext+0avgdata 38076maxresident)k
1216inputs+1250616outputs (8major+10881minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480157_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480157_2_p.fastq -S SRR6480157.sam > SRR6480157_1_p_fastq_SRR6480157_1_p_fastq__bowtie2_log
369924 reads; of these:
  369924 (100.00%) were paired; of these:
    112309 (30.36%) aligned concordantly 0 times
    251202 (67.91%) aligned concordantly exactly 1 time
    6413 (1.73%) aligned concordantly >1 times
    ----
    112309 pairs aligned concordantly 0 times; of these:
      84256 (75.02%) aligned discordantly 1 time
    ----
    28053 pairs aligned 0 times concordantly or discordantly; of these:
      56106 mates make up the pairs; of these:
        50260 (89.58%) aligned 0 times
        2085 (3.72%) aligned exactly 1 time
        3761 (6.70%) aligned >1 times
93.21% overall alignment rate
197.69user 2.56system 3:20.02elapsed 100%CPU (0avgtext+0avgdata 37744maxresident)k
0inputs+790488outputs (0major+10785minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480146_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480146_2_p.fastq -S SRR6480146.sam > SRR6480146_1_p_fastq_SRR6480146_1_p_fastq__bowtie2_log
687510 reads; of these:
  687510 (100.00%) were paired; of these:
    119101 (17.32%) aligned concordantly 0 times
    555222 (80.76%) aligned concordantly exactly 1 time
    13187 (1.92%) aligned concordantly >1 times
    ----
    119101 pairs aligned concordantly 0 times; of these:
      74147 (62.26%) aligned discordantly 1 time
    ----
    44954 pairs aligned 0 times concordantly or discordantly; of these:
      89908 mates make up the pairs; of these:
        84649 (94.15%) aligned 0 times
        1984 (2.21%) aligned exactly 1 time
        3275 (3.64%) aligned >1 times
93.84% overall alignment rate
316.19user 4.55system 5:20.38elapsed 100%CPU (0avgtext+0avgdata 37880maxresident)k
0inputs+1332696outputs (0major+10896minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480144_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480144_2_p.fastq -S SRR6480144.sam > SRR6480144_1_p_fastq_SRR6480144_1_p_fastq__bowtie2_log
442191 reads; of these:
  442191 (100.00%) were paired; of these:
    106737 (24.14%) aligned concordantly 0 times
    327144 (73.98%) aligned concordantly exactly 1 time
    8310 (1.88%) aligned concordantly >1 times
    ----
    106737 pairs aligned concordantly 0 times; of these:
      72846 (68.25%) aligned discordantly 1 time
    ----
    33891 pairs aligned 0 times concordantly or discordantly; of these:
      67782 mates make up the pairs; of these:
        62462 (92.15%) aligned 0 times
        2035 (3.00%) aligned exactly 1 time
        3285 (4.85%) aligned >1 times
92.94% overall alignment rate
228.15user 2.86system 3:50.74elapsed 100%CPU (0avgtext+0avgdata 38132maxresident)k
0inputs+924824outputs (0major+10862minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480136_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480136_2_p.fastq -S SRR6480136.sam > SRR6480136_1_p_fastq_SRR6480136_1_p_fastq__bowtie2_log
410096 reads; of these:
  410096 (100.00%) were paired; of these:
    140938 (34.37%) aligned concordantly 0 times
    261169 (63.68%) aligned concordantly exactly 1 time
    7989 (1.95%) aligned concordantly >1 times
    ----
    140938 pairs aligned concordantly 0 times; of these:
      72532 (51.46%) aligned discordantly 1 time
    ----
    68406 pairs aligned 0 times concordantly or discordantly; of these:
      136812 mates make up the pairs; of these:
        128756 (94.11%) aligned 0 times
        3775 (2.76%) aligned exactly 1 time
        4281 (3.13%) aligned >1 times
84.30% overall alignment rate
226.84user 2.80system 3:49.39elapsed 100%CPU (0avgtext+0avgdata 37116maxresident)k
0inputs+851976outputs (0major+10660minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480153_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480153_2_p.fastq -S SRR6480153.sam > SRR6480153_1_p_fastq_SRR6480153_1_p_fastq__bowtie2_log
432722 reads; of these:
  432722 (100.00%) were paired; of these:
    177551 (41.03%) aligned concordantly 0 times
    246463 (56.96%) aligned concordantly exactly 1 time
    8708 (2.01%) aligned concordantly >1 times
    ----
    177551 pairs aligned concordantly 0 times; of these:
      104378 (58.79%) aligned discordantly 1 time
    ----
    73173 pairs aligned 0 times concordantly or discordantly; of these:
      146346 mates make up the pairs; of these:
        132846 (90.78%) aligned 0 times
        6085 (4.16%) aligned exactly 1 time
        7415 (5.07%) aligned >1 times
84.65% overall alignment rate
254.44user 2.17system 4:16.32elapsed 100%CPU (0avgtext+0avgdata 38320maxresident)k
0inputs+922336outputs (0major+10890minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480156_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480156_2_p.fastq -S SRR6480156.sam > SRR6480156_1_p_fastq_SRR6480156_1_p_fastq__bowtie2_log
389521 reads; of these:
  389521 (100.00%) were paired; of these:
    152811 (39.23%) aligned concordantly 0 times
    230572 (59.19%) aligned concordantly exactly 1 time
    6138 (1.58%) aligned concordantly >1 times
    ----
    152811 pairs aligned concordantly 0 times; of these:
      78900 (51.63%) aligned discordantly 1 time
    ----
    73911 pairs aligned 0 times concordantly or discordantly; of these:
      147822 mates make up the pairs; of these:
        138728 (93.85%) aligned 0 times
        4944 (3.34%) aligned exactly 1 time
        4150 (2.81%) aligned >1 times
82.19% overall alignment rate
215.59user 2.45system 3:37.78elapsed 100%CPU (0avgtext+0avgdata 38008maxresident)k
0inputs+816576outputs (0major+10834minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480155_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480155_2_p.fastq -S SRR6480155.sam > SRR6480155_1_p_fastq_SRR6480155_1_p_fastq__bowtie2_log
516472 reads; of these:
  516472 (100.00%) were paired; of these:
    145750 (28.22%) aligned concordantly 0 times
    361114 (69.92%) aligned concordantly exactly 1 time
    9608 (1.86%) aligned concordantly >1 times
    ----
    145750 pairs aligned concordantly 0 times; of these:
      70923 (48.66%) aligned discordantly 1 time
    ----
    74827 pairs aligned 0 times concordantly or discordantly; of these:
      149654 mates make up the pairs; of these:
        142675 (95.34%) aligned 0 times
        3443 (2.30%) aligned exactly 1 time
        3536 (2.36%) aligned >1 times
86.19% overall alignment rate
283.47user 3.12system 4:46.27elapsed 100%CPU (0avgtext+0avgdata 37608maxresident)k
0inputs+1079920outputs (0major+10760minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480141_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480141_2_p.fastq -S SRR6480141.sam > SRR6480141_1_p_fastq_SRR6480141_1_p_fastq__bowtie2_log
450596 reads; of these:
  450596 (100.00%) were paired; of these:
    122650 (27.22%) aligned concordantly 0 times
    320800 (71.19%) aligned concordantly exactly 1 time
    7146 (1.59%) aligned concordantly >1 times
    ----
    122650 pairs aligned concordantly 0 times; of these:
      95557 (77.91%) aligned discordantly 1 time
    ----
    27093 pairs aligned 0 times concordantly or discordantly; of these:
      54186 mates make up the pairs; of these:
        47846 (88.30%) aligned 0 times
        2276 (4.20%) aligned exactly 1 time
        4064 (7.50%) aligned >1 times
94.69% overall alignment rate
237.81user 3.13system 4:00.69elapsed 100%CPU (0avgtext+0avgdata 38016maxresident)k
0inputs+962488outputs (0major+10911minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480147_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480147_2_p.fastq -S SRR6480147.sam > SRR6480147_1_p_fastq_SRR6480147_1_p_fastq__bowtie2_log
663572 reads; of these:
  663572 (100.00%) were paired; of these:
    224814 (33.88%) aligned concordantly 0 times
    425123 (64.07%) aligned concordantly exactly 1 time
    13635 (2.05%) aligned concordantly >1 times
    ----
    224814 pairs aligned concordantly 0 times; of these:
      82800 (36.83%) aligned discordantly 1 time
    ----
    142014 pairs aligned 0 times concordantly or discordantly; of these:
      284028 mates make up the pairs; of these:
        273885 (96.43%) aligned 0 times
        5220 (1.84%) aligned exactly 1 time
        4923 (1.73%) aligned >1 times
79.36% overall alignment rate
283.73user 3.79system 4:47.12elapsed 100%CPU (0avgtext+0avgdata 38576maxresident)k
0inputs+1180720outputs (0major+10993minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480138_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480138_2_p.fastq -S SRR6480138.sam > SRR6480138_1_p_fastq_SRR6480138_1_p_fastq__bowtie2_log
466115 reads; of these:
  466115 (100.00%) were paired; of these:
    183102 (39.28%) aligned concordantly 0 times
    272477 (58.46%) aligned concordantly exactly 1 time
    10536 (2.26%) aligned concordantly >1 times
    ----
    183102 pairs aligned concordantly 0 times; of these:
      86402 (47.19%) aligned discordantly 1 time
    ----
    96700 pairs aligned 0 times concordantly or discordantly; of these:
      193400 mates make up the pairs; of these:
        180264 (93.21%) aligned 0 times
        6226 (3.22%) aligned exactly 1 time
        6910 (3.57%) aligned >1 times
80.66% overall alignment rate
260.68user 3.48system 4:23.87elapsed 100%CPU (0avgtext+0avgdata 37784maxresident)k
0inputs+985848outputs (0major+10753minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480149_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480149_2_p.fastq -S SRR6480149.sam > SRR6480149_1_p_fastq_SRR6480149_1_p_fastq__bowtie2_log
597270 reads; of these:
  597270 (100.00%) were paired; of these:
    196577 (32.91%) aligned concordantly 0 times
    389021 (65.13%) aligned concordantly exactly 1 time
    11672 (1.95%) aligned concordantly >1 times
    ----
    196577 pairs aligned concordantly 0 times; of these:
      92851 (47.23%) aligned discordantly 1 time
    ----
    103726 pairs aligned 0 times concordantly or discordantly; of these:
      207452 mates make up the pairs; of these:
        197613 (95.26%) aligned 0 times
        4241 (2.04%) aligned exactly 1 time
        5598 (2.70%) aligned >1 times
83.46% overall alignment rate
285.83user 3.83system 4:49.34elapsed 100%CPU (0avgtext+0avgdata 38328maxresident)k
0inputs+1130224outputs (0major+10942minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480145_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480145_2_p.fastq -S SRR6480145.sam > SRR6480145_1_p_fastq_SRR6480145_1_p_fastq__bowtie2_log
583876 reads; of these:
  583876 (100.00%) were paired; of these:
    141066 (24.16%) aligned concordantly 0 times
    431761 (73.95%) aligned concordantly exactly 1 time
    11049 (1.89%) aligned concordantly >1 times
    ----
    141066 pairs aligned concordantly 0 times; of these:
      107284 (76.05%) aligned discordantly 1 time
    ----
    33782 pairs aligned 0 times concordantly or discordantly; of these:
      67564 mates make up the pairs; of these:
        60143 (89.02%) aligned 0 times
        2556 (3.78%) aligned exactly 1 time
        4865 (7.20%) aligned >1 times
94.85% overall alignment rate
308.96user 4.35system 5:12.96elapsed 100%CPU (0avgtext+0avgdata 38084maxresident)k
0inputs+1239016outputs (0major+10925minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480140_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480140_2_p.fastq -S SRR6480140.sam > SRR6480140_1_p_fastq_SRR6480140_1_p_fastq__bowtie2_log
475991 reads; of these:
  475991 (100.00%) were paired; of these:
    152721 (32.08%) aligned concordantly 0 times
    315155 (66.21%) aligned concordantly exactly 1 time
    8115 (1.70%) aligned concordantly >1 times
    ----
    152721 pairs aligned concordantly 0 times; of these:
      117257 (76.78%) aligned discordantly 1 time
    ----
    35464 pairs aligned 0 times concordantly or discordantly; of these:
      70928 mates make up the pairs; of these:
        62795 (88.53%) aligned 0 times
        2832 (3.99%) aligned exactly 1 time
        5301 (7.47%) aligned >1 times
93.40% overall alignment rate
256.17user 2.93system 4:18.82elapsed 100%CPU (0avgtext+0avgdata 38420maxresident)k
0inputs+1015264outputs (0major+11012minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480151_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480151_2_p.fastq -S SRR6480151.sam > SRR6480151_1_p_fastq_SRR6480151_1_p_fastq__bowtie2_log
675510 reads; of these:
  675510 (100.00%) were paired; of these:
    162831 (24.10%) aligned concordantly 0 times
    499490 (73.94%) aligned concordantly exactly 1 time
    13189 (1.95%) aligned concordantly >1 times
    ----
    162831 pairs aligned concordantly 0 times; of these:
      91590 (56.25%) aligned discordantly 1 time
    ----
    71241 pairs aligned 0 times concordantly or discordantly; of these:
      142482 mates make up the pairs; of these:
        134224 (94.20%) aligned 0 times
        3828 (2.69%) aligned exactly 1 time
        4430 (3.11%) aligned >1 times
90.06% overall alignment rate
304.90user 3.96system 5:08.56elapsed 100%CPU (0avgtext+0avgdata 38684maxresident)k
0inputs+1272912outputs (0major+10983minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480139_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480139_2_p.fastq -S SRR6480139.sam > SRR6480139_1_p_fastq_SRR6480139_1_p_fastq__bowtie2_log
594393 reads; of these:
  594393 (100.00%) were paired; of these:
    174969 (29.44%) aligned concordantly 0 times
    406343 (68.36%) aligned concordantly exactly 1 time
    13081 (2.20%) aligned concordantly >1 times
    ----
    174969 pairs aligned concordantly 0 times; of these:
      85084 (48.63%) aligned discordantly 1 time
    ----
    89885 pairs aligned 0 times concordantly or discordantly; of these:
      179770 mates make up the pairs; of these:
        169462 (94.27%) aligned 0 times
        4896 (2.72%) aligned exactly 1 time
        5412 (3.01%) aligned >1 times
85.74% overall alignment rate
320.16user 4.07system 5:23.87elapsed 100%CPU (0avgtext+0avgdata 38080maxresident)k
0inputs+1215648outputs (0major+10840minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480143_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480143_2_p.fastq -S SRR6480143.sam > SRR6480143_1_p_fastq_SRR6480143_1_p_fastq__bowtie2_log
440028 reads; of these:
  440028 (100.00%) were paired; of these:
    103420 (23.50%) aligned concordantly 0 times
    328708 (74.70%) aligned concordantly exactly 1 time
    7900 (1.80%) aligned concordantly >1 times
    ----
    103420 pairs aligned concordantly 0 times; of these:
      78045 (75.46%) aligned discordantly 1 time
    ----
    25375 pairs aligned 0 times concordantly or discordantly; of these:
      50750 mates make up the pairs; of these:
        45271 (89.20%) aligned 0 times
        2059 (4.06%) aligned exactly 1 time
        3420 (6.74%) aligned >1 times
94.86% overall alignment rate
224.89user 2.89system 3:47.53elapsed 100%CPU (0avgtext+0avgdata 36752maxresident)k
0inputs+911304outputs (0major+10629minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480148_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480148_2_p.fastq -S SRR6480148.sam > SRR6480148_1_p_fastq_SRR6480148_1_p_fastq__bowtie2_log
473610 reads; of these:
  473610 (100.00%) were paired; of these:
    137746 (29.08%) aligned concordantly 0 times
    326308 (68.90%) aligned concordantly exactly 1 time
    9556 (2.02%) aligned concordantly >1 times
    ----
    137746 pairs aligned concordantly 0 times; of these:
      65350 (47.44%) aligned discordantly 1 time
    ----
    72396 pairs aligned 0 times concordantly or discordantly; of these:
      144792 mates make up the pairs; of these:
        137183 (94.74%) aligned 0 times
        3824 (2.64%) aligned exactly 1 time
        3785 (2.61%) aligned >1 times
85.52% overall alignment rate
259.65user 3.69system 4:23.07elapsed 100%CPU (0avgtext+0avgdata 38596maxresident)k
0inputs+989480outputs (0major+11126minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480154_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480154_2_p.fastq -S SRR6480154.sam > SRR6480154_1_p_fastq_SRR6480154_1_p_fastq__bowtie2_log
777108 reads; of these:
  777108 (100.00%) were paired; of these:
    96939 (12.47%) aligned concordantly 0 times
    661918 (85.18%) aligned concordantly exactly 1 time
    18251 (2.35%) aligned concordantly >1 times
    ----
    96939 pairs aligned concordantly 0 times; of these:
      37803 (39.00%) aligned discordantly 1 time
    ----
    59136 pairs aligned 0 times concordantly or discordantly; of these:
      118272 mates make up the pairs; of these:
        114410 (96.73%) aligned 0 times
        1935 (1.64%) aligned exactly 1 time
        1927 (1.63%) aligned >1 times
92.64% overall alignment rate
335.91user 4.87system 5:40.66elapsed 100%CPU (0avgtext+0avgdata 37584maxresident)k
832inputs+1463704outputs (7major+10759minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480137_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480137_2_p.fastq -S SRR6480137.sam > SRR6480137_1_p_fastq_SRR6480137_1_p_fastq__bowtie2_log
465844 reads; of these:
  465844 (100.00%) were paired; of these:
    132259 (28.39%) aligned concordantly 0 times
    325044 (69.78%) aligned concordantly exactly 1 time
    8541 (1.83%) aligned concordantly >1 times
    ----
    132259 pairs aligned concordantly 0 times; of these:
      89229 (67.47%) aligned discordantly 1 time
    ----
    43030 pairs aligned 0 times concordantly or discordantly; of these:
      86060 mates make up the pairs; of these:
        78213 (90.88%) aligned 0 times
        3357 (3.90%) aligned exactly 1 time
        4490 (5.22%) aligned >1 times
91.61% overall alignment rate
249.12user 2.76system 4:11.59elapsed 100%CPU (0avgtext+0avgdata 37692maxresident)k
0inputs+988360outputs (0major+10818minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480142_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480142_2_p.fastq -S SRR6480142.sam > SRR6480142_1_p_fastq_SRR6480142_1_p_fastq__bowtie2_log
404494 reads; of these:
  404494 (100.00%) were paired; of these:
    104709 (25.89%) aligned concordantly 0 times
    291497 (72.06%) aligned concordantly exactly 1 time
    8288 (2.05%) aligned concordantly >1 times
    ----
    104709 pairs aligned concordantly 0 times; of these:
      71899 (68.67%) aligned discordantly 1 time
    ----
    32810 pairs aligned 0 times concordantly or discordantly; of these:
      65620 mates make up the pairs; of these:
        60002 (91.44%) aligned 0 times
        1891 (2.88%) aligned exactly 1 time
        3727 (5.68%) aligned >1 times
92.58% overall alignment rate
209.82user 2.33system 3:31.91elapsed 100%CPU (0avgtext+0avgdata 37880maxresident)k
0inputs+854720outputs (0major+10820minor)pagefaults 0swaps
Executing.. 
time bowtie2 -x NZ_CP032222 -q -1 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480152_1_p.fastq -2 /home/ubuntu/bioinf/Klebsiella/10_other_samples/trimmomatic/SRR6480152_2_p.fastq -S SRR6480152.sam > SRR6480152_1_p_fastq_SRR6480152_1_p_fastq__bowtie2_log
750734 reads; of these:
  750734 (100.00%) were paired; of these:
    248941 (33.16%) aligned concordantly 0 times
    484447 (64.53%) aligned concordantly exactly 1 time
    17346 (2.31%) aligned concordantly >1 times
    ----
    248941 pairs aligned concordantly 0 times; of these:
      100263 (40.28%) aligned discordantly 1 time
    ----
    148678 pairs aligned 0 times concordantly or discordantly; of these:
      297356 mates make up the pairs; of these:
        282173 (94.89%) aligned 0 times
        7139 (2.40%) aligned exactly 1 time
        8044 (2.71%) aligned >1 times
81.21% overall alignment rate
328.55user 4.41system 5:32.65elapsed 100%CPU (0avgtext+0avgdata 38884maxresident)k
0inputs+1419896outputs (0major+11083minor)pagefaults 0swaps
```

