# KPN1053 alignments (2018-10-22)

## Overview
In this part the following steps were executed:
  - Download ~ 200 references, split them to chromosomes and plasmids
  - Select the best chromosome reference using RefRank tool
  - Align Klebsiella reads to selected chromosome reference
  - Analyze alignment quality
  - Try to find plasmids using PlasmidFinder against raw reads and against assembled Klebsiella genome

See all results here: http://195.208.116.225/bioinf/Klebsiella/9_alignment/

## Installation requirements
See https://github.com/dariader/Klebsiella/blob/alignments/README.md

## 1. Data preperation

You can find more instructions here: https://www.ncbi.nlm.nih.gov/genome/doc/ftpfaq/

### 1.1 Find and download all available Klebsiella pneumaniae genomes

First of all, we want to download all available complete Klebsiella pneumaniae genomes.
This command will download NCBI assembly summary, which contains all available bacteria genomes:
```
wget ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt
```

Then we can find all Klebsiella pneumaniae complete genomes:
```
awk -F "\t" '$8=="Klebsiella pneumoniae" &&  $12=="Complete Genome" && $11=="latest" {print $20}' assembly_summary.txt > klebsiella_pneumaniae_genomes.txt
```

Let's calculate a number of found genomes:
```
wc -l klebsiella_pneumaniae_genomes.txt 

212 klebsiella_pneumaniae_genomes.txt
```

Construct full FTP paths to selected genomes:
```
awk 'BEGIN { FS=OFS="/"; filesuffix="genomic.fna.gz" }{ ftpdir=$0;asm=$10;file=asm"_"filesuffix; print ftpdir,file }' klebsiella_pneumaniae_genomes.txt > klebsiella_pneumaniae_genome_paths.txt
```

Create a directory for Klebsiella genomes:

```
 mkdir klebsiella_reference
 cd ./klebsiella_reference/
```

// TODO: download GFF files for best genome later
Download all selected genomes:
```
time cat ../klebsiella_pneumaniae_genome_paths.txt | parallel --gnu "wget {}"

real	45m28.196s
user	0m2.496s
sys	0m6.151s
```

Unzip genomes:
```
gunzip ../../data/klebsiella_reference/*.fna.gz
```

### 1.2 Split genomes to chromosomes and plasmids

Split genomes to per-sequence files using faidx tool:
```
time ls ../../../data/klebsiella_reference/ | xargs -n 1 basename | parallel --gnu "faidx -x ../../../data/klebsiella_reference/{}"
```

To figure out, which files are chromosomes and which are plasmids, we can use assembly report from NCBI.

Create a directory for assembly reports:
```
mkdir klebsiella_assembly_reports
```

Create full FTP paths to assembly reports:
```
awk 'BEGIN { FS=OFS="/"; filesuffix="assembly_report.txt" }{ ftpdir=$0;asm=$10;file=asm"_"filesuffix; print ftpdir,file }' klebsiella_pneumaniae_genomes.txt > klebsiella_pneumaniae_genome_reports.txt

wc -l klebsiella_pneumaniae_genome_reports.txt
212 klebsiella_pneumaniae_genome_reports.txt
```

Download all assembly reports:
```
time cat ../klebsiella_pneumaniae_genome_reports.txt | parallel --gnu "wget {}"

real	60m39.766s
user	0m2.153s
sys	0m3.175s
```

Select all chromosome names and plasmid names into separete files:
```
awk -F '\t' '{if ($4 =="Chromosome") print $7}' klebsiella_assembly_reports/GCF_* > klebsiella_chromosomes.txt
awk -F '\t' '{if ($4 =="Plasmid") print $7}' klebsiella_assembly_reports/GCF_* > klebsiella_plasmids.txt

wc -l klebsiella_chromosomes.txt
212 klebsiella_chromosomes.txt

wc -l klebsiella_plasmids.txt
728 klebsiella_plasmids.txt
```

Construct full paths to chromosome and plasmid file names:
```
awk 'BEGIN { fileprefix="/home/ubuntu/bioinf/Klebsiella/9_alignment/single_sequences" }{ chr=$1;file=fileprefix"/"chr".fna" ; print file }' ../../data/klebsiella_chromosomes.txt > klebsiella_chromosome_paths.txt

awk 'BEGIN { fileprefix="/home/ubuntu/bioinf/Klebsiella/9_alignment/single_sequences" }{ plsmd=$1;file=fileprefix"/"plsmd".fna" ; print file }' ../../data/klebsiella_plasmids.txt > klebsiella_plasmid_paths.txt
```

## 2. Select best chromosome reference
### 2.1 Run refRank against list of chromosome files

Run RefRank using chromosome sequences as target refernce files:
```
time ../../software/refRank/refrank.py --fastq ../2_kpn1053_trimmomatic/2/SRR6480154_1_P.fastq ../2_kpn1053_trimmomatic/2/SRR6480154_2_P.fastq --ref $(cat klebsiella_chromosome_paths.txt)
```
