# KPN1053 alignments (2018-10-21)

In this part the following steps were executed:
  - Filtering out plasmid sequences
  - Rerun RefRank againt references
  - Align reads to the best reference

See all results here: http://195.208.116.225/bioinf/Klebsiella/8_alignment_without_plasmids/

## Installation requirements
See https://github.com/dariader/Klebsiella/blob/alignments/README.md

## 1. Data preperation

### 1.1 Filter out plasmid sequences

Using assembly search (https://www.ncbi.nlm.nih.gov/assembly/), find all chromosomes from splitted fasta files.
Chromosomes:
- 	CP031885.1 (https://www.ncbi.nlm.nih.gov/assembly/GCF_001887995.2/)
- 	CP018428.1 (https://www.ncbi.nlm.nih.gov/assembly/GCF_001902195.1/)
- 	CP031800.1 (https://www.ncbi.nlm.nih.gov/assembly/GCF_003432165.1/)
- 	CP032207.1 (https://www.ncbi.nlm.nih.gov/assembly/GCF_003571705.1/)

## 2. Reference selection

### 2.1 Run refRank against choromosome references 

```
time refrank.py --fastq SRR6480154_1_P.fastq SRR6480154_2_P.fastq --ref NZ_CP031885.1.fna NZ_CP018428.1.fna NZ_CP031800.1.fna NZ_CP032207.1.fna

preparing output folder ...
  REFRANK_2018-10-21_18-01-22
processing reference files ... [4/4]       
processing fastq files ... [2/2]       
indexing reference files ... [4/4]       
preparing next dataset...
  paired-end read data (forward): SRR6480154_1_P.fastq
  paired-end read data (reverse): SRR6480154_2_P.fastq
  read count: 777108 per file
  downsampling ...
    SRR6480154_1_P.fastq
    SRR6480154_2_P.fastq
    fraction: 77710 reads per file
TASK 1 OF 4
  mapping...
  analyzing...
  norm_avg_base_cov: 2.874e-05
TASK 2 OF 4
  mapping...
  analyzing...
  norm_avg_base_cov: 2.709e-05
TASK 3 OF 4
  mapping...
  analyzing...
  norm_avg_base_cov: 2.736e-05
TASK 4 OF 4
  mapping...
  analyzing...
  norm_avg_base_cov: 2.842e-05
writing result report ...

real	0m55.708s
user	2m28.715s
sys	0m10.884s
```

## 3. Alignment

### 3.1 Index chromosome

```sh
time bowtie2-build ~/bioinf/data/reference/NZ_CP031885.1.fna NZ_CP031885
Settings:
  Output files: "NZ_CP031885.*.bt2"
  Line rate: 6 (line is 64 bytes)
  Lines per side: 1 (side is 64 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  /home/ubuntu/bioinf/data/reference/NZ_CP031885.1.fna
Building a SMALL index
Reading reference sizes
  Time reading reference sizes: 00:00:00
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
bmax according to bmaxDivN setting: 1309981
Using parameters --bmax 982486 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 982486 --dcv 1024
Constructing suffix-array element generator
Building DifferenceCoverSample
  Building sPrime
  Building sPrimeOrder
  V-Sorting samples
  V-Sorting samples time: 00:00:00
  Allocating rank array
  Ranking v-sort output
  Ranking v-sort output time: 00:00:00
  Invoking Larsson-Sadakane on ranks
  Invoking Larsson-Sadakane on ranks time: 00:00:00
  Sanity-checking and returning
Building samples
Reserving space for 12 sample suffixes
Generating random suffixes
QSorting 12 sample offsets, eliminating duplicates
QSorting sample offsets, eliminating duplicates time: 00:00:00
Multikey QSorting 12 samples
  (Using difference cover)
  Multikey QSorting samples time: 00:00:00
Calculating bucket sizes
Splitting and merging
  Splitting and merging time: 00:00:00
Avg bucket size: 5.23992e+06 (target: 982485)
Converting suffix-array elements to index image
Allocating ftab, absorbFtab
Entering Ebwt loop
Getting block 1 of 1
  No samples; assembling all-inclusive block
  Sorting block of length 5239924 for bucket 1
  (Using difference cover)
  Sorting block time: 00:00:01
Returning block of 5239925 for bucket 1
Exited Ebwt loop
fchr[A]: 0
fchr[C]: 1113390
fchr[G]: 2619949
fchr[T]: 4127802
fchr[$]: 5239924
Exiting Ebwt::buildToDisk()
Returning from initFromVector
Wrote 5941167 bytes to primary EBWT file: NZ_CP031885.1.bt2
Wrote 1309988 bytes to secondary EBWT file: NZ_CP031885.2.bt2
Re-opening _in1 and _in2 as input streams
Returning from Ebwt constructor
Headers:
    len: 5239924
    bwtLen: 5239925
    sz: 1309981
    bwtSz: 1309982
    lineRate: 6
    offRate: 4
    offMask: 0xfffffff0
    ftabChars: 10
    eftabLen: 20
    eftabSz: 80
    ftabLen: 1048577
    ftabSz: 4194308
    offsLen: 327496
    offsSz: 1309984
    lineSz: 64
    sideSz: 64
    sideBwtSz: 48
    sideBwtLen: 192
    numSides: 27292
    numLines: 27292
    ebwtTotLen: 1746688
    ebwtTotSz: 1746688
    color: 0
    reverse: 0
Total time for call to driver() for forward index: 00:00:01
Reading reference sizes
  Time reading reference sizes: 00:00:00
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
  Time to reverse reference sequence: 00:00:00
bmax according to bmaxDivN setting: 1309981
Using parameters --bmax 982486 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 982486 --dcv 1024
Constructing suffix-array element generator
Building DifferenceCoverSample
  Building sPrime
  Building sPrimeOrder
  V-Sorting samples
  V-Sorting samples time: 00:00:00
  Allocating rank array
  Ranking v-sort output
  Ranking v-sort output time: 00:00:00
  Invoking Larsson-Sadakane on ranks
  Invoking Larsson-Sadakane on ranks time: 00:00:00
  Sanity-checking and returning
Building samples
Reserving space for 12 sample suffixes
Generating random suffixes
QSorting 12 sample offsets, eliminating duplicates
QSorting sample offsets, eliminating duplicates time: 00:00:00
Multikey QSorting 12 samples
  (Using difference cover)
  Multikey QSorting samples time: 00:00:00
Calculating bucket sizes
Splitting and merging
  Splitting and merging time: 00:00:00
Avg bucket size: 5.23992e+06 (target: 982485)
Converting suffix-array elements to index image
Allocating ftab, absorbFtab
Entering Ebwt loop
Getting block 1 of 1
  No samples; assembling all-inclusive block
  Sorting block of length 5239924 for bucket 1
  (Using difference cover)
  Sorting block time: 00:00:01
Returning block of 5239925 for bucket 1
Exited Ebwt loop
fchr[A]: 0
fchr[C]: 1113390
fchr[G]: 2619949
fchr[T]: 4127802
fchr[$]: 5239924
Exiting Ebwt::buildToDisk()
Returning from initFromVector
Wrote 5941167 bytes to primary EBWT file: NZ_CP031885.rev.1.bt2
Wrote 1309988 bytes to secondary EBWT file: NZ_CP031885.rev.2.bt2
Re-opening _in1 and _in2 as input streams
Returning from Ebwt constructor
Headers:
    len: 5239924
    bwtLen: 5239925
    sz: 1309981
    bwtSz: 1309982
    lineRate: 6
    offRate: 4
    offMask: 0xfffffff0
    ftabChars: 10
    eftabLen: 20
    eftabSz: 80
    ftabLen: 1048577
    ftabSz: 4194308
    offsLen: 327496
    offsSz: 1309984
    lineSz: 64
    sideSz: 64
    sideBwtSz: 48
    sideBwtLen: 192
    numSides: 27292
    numLines: 27292
    ebwtTotLen: 1746688
    ebwtTotSz: 1746688
    color: 0
    reverse: 1
Total time for backward call to driver() for mirror index: 00:00:02

real	0m3.303s
user	0m2.970s
sys	0m0.269s
```

### 3.2 Align against choromosome

```sh
time bowtie2 -x NZ_CP031885 -q -1 SRR6480154_1_P.fastq -2 SRR6480154_2_P.fastq -S NZ_CP031885.sam

777108 reads; of these:
  777108 (100.00%) were paired; of these:
    150990 (19.43%) aligned concordantly 0 times
    615590 (79.22%) aligned concordantly exactly 1 time
    10528 (1.35%) aligned concordantly >1 times
    ----
    150990 pairs aligned concordantly 0 times; of these:
      35727 (23.66%) aligned discordantly 1 time
    ----
    115263 pairs aligned 0 times concordantly or discordantly; of these:
      230526 mates make up the pairs; of these:
        226922 (98.44%) aligned 0 times
        2600 (1.13%) aligned exactly 1 time
        1004 (0.44%) aligned >1 times
85.40% overall alignment rate

real	5m47.518s
user	5m43.307s
sys	0m4.558s
```

### 3.3 Index whole reference

```sh
time bowtie2-build GCF_001887995.2_ASM188799v2_genomic.fna GCF_001887995
Settings:
  Output files: "GCF_001887995.*.bt2"
  Line rate: 6 (line is 64 bytes)
  Lines per side: 1 (side is 64 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  /home/ubuntu/bioinf/data/reference/GCF_001887995.2_ASM188799v2_genomic.fna
Building a SMALL index
Reading reference sizes
  Time reading reference sizes: 00:00:00
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
bmax according to bmaxDivN setting: 1365470
Using parameters --bmax 1024103 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 1024103 --dcv 1024
Constructing suffix-array element generator
Building DifferenceCoverSample
  Building sPrime
  Building sPrimeOrder
  V-Sorting samples
  V-Sorting samples time: 00:00:00
  Allocating rank array
  Ranking v-sort output
  Ranking v-sort output time: 00:00:00
  Invoking Larsson-Sadakane on ranks
  Invoking Larsson-Sadakane on ranks time: 00:00:00
  Sanity-checking and returning
Building samples
Reserving space for 12 sample suffixes
Generating random suffixes
QSorting 12 sample offsets, eliminating duplicates
QSorting sample offsets, eliminating duplicates time: 00:00:00
Multikey QSorting 12 samples
  (Using difference cover)
  Multikey QSorting samples time: 00:00:00
Calculating bucket sizes
Splitting and merging
  Splitting and merging time: 00:00:00
Avg bucket size: 5.46188e+06 (target: 1024102)
Converting suffix-array elements to index image
Allocating ftab, absorbFtab
Entering Ebwt loop
Getting block 1 of 1
  No samples; assembling all-inclusive block
  Sorting block of length 5461882 for bucket 1
  (Using difference cover)
  Sorting block time: 00:00:01
Returning block of 5461883 for bucket 1
Exited Ebwt loop
fchr[A]: 0
fchr[C]: 1166732
fchr[G]: 2730702
fchr[T]: 4295277
fchr[$]: 5461882
Exiting Ebwt::buildToDisk()
Returning from initFromVector
Wrote 6015550 bytes to primary EBWT file: GCF_001887995.1.bt2
Wrote 1365476 bytes to secondary EBWT file: GCF_001887995.2.bt2
Re-opening _in1 and _in2 as input streams
Returning from Ebwt constructor
Headers:
    len: 5461882
    bwtLen: 5461883
    sz: 1365471
    bwtSz: 1365471
    lineRate: 6
    offRate: 4
    offMask: 0xfffffff0
    ftabChars: 10
    eftabLen: 20
    eftabSz: 80
    ftabLen: 1048577
    ftabSz: 4194308
    offsLen: 341368
    offsSz: 1365472
    lineSz: 64
    sideSz: 64
    sideBwtSz: 48
    sideBwtLen: 192
    numSides: 28448
    numLines: 28448
    ebwtTotLen: 1820672
    ebwtTotSz: 1820672
    color: 0
    reverse: 0
Total time for call to driver() for forward index: 00:00:02
Reading reference sizes
  Time reading reference sizes: 00:00:00
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
  Time to reverse reference sequence: 00:00:00
bmax according to bmaxDivN setting: 1365470
Using parameters --bmax 1024103 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 1024103 --dcv 1024
Constructing suffix-array element generator
Building DifferenceCoverSample
  Building sPrime
  Building sPrimeOrder
  V-Sorting samples
  V-Sorting samples time: 00:00:00
  Allocating rank array
  Ranking v-sort output
  Ranking v-sort output time: 00:00:00
  Invoking Larsson-Sadakane on ranks
  Invoking Larsson-Sadakane on ranks time: 00:00:00
  Sanity-checking and returning
Building samples
Reserving space for 12 sample suffixes
Generating random suffixes
QSorting 12 sample offsets, eliminating duplicates
QSorting sample offsets, eliminating duplicates time: 00:00:00
Multikey QSorting 12 samples
  (Using difference cover)
  Multikey QSorting samples time: 00:00:00
Calculating bucket sizes
Splitting and merging
  Splitting and merging time: 00:00:00
Avg bucket size: 5.46188e+06 (target: 1024102)
Converting suffix-array elements to index image
Allocating ftab, absorbFtab
Entering Ebwt loop
Getting block 1 of 1
  No samples; assembling all-inclusive block
  Sorting block of length 5461882 for bucket 1
  (Using difference cover)
  Sorting block time: 00:00:01
Returning block of 5461883 for bucket 1
Exited Ebwt loop
fchr[A]: 0
fchr[C]: 1166732
fchr[G]: 2730702
fchr[T]: 4295277
fchr[$]: 5461882
Exiting Ebwt::buildToDisk()
Returning from initFromVector
Wrote 6015550 bytes to primary EBWT file: GCF_001887995.rev.1.bt2
Wrote 1365476 bytes to secondary EBWT file: GCF_001887995.rev.2.bt2
Re-opening _in1 and _in2 as input streams
Returning from Ebwt constructor
Headers:
    len: 5461882
    bwtLen: 5461883
    sz: 1365471
    bwtSz: 1365471
    lineRate: 6
    offRate: 4
    offMask: 0xfffffff0
    ftabChars: 10
    eftabLen: 20
    eftabSz: 80
    ftabLen: 1048577
    ftabSz: 4194308
    offsLen: 341368
    offsSz: 1365472
    lineSz: 64
    sideSz: 64
    sideBwtSz: 48
    sideBwtLen: 192
    numSides: 28448
    numLines: 28448
    ebwtTotLen: 1820672
    ebwtTotSz: 1820672
    color: 0
    reverse: 1
Total time for backward call to driver() for mirror index: 00:00:01

real	0m3.496s
user	0m3.110s
sys	0m0.386s
```

### 3.4 Align against whole reference

```sh
time bowtie2 -x GCF_001887995 -q -1 SRR6480154_1_P.fastq -2 SRR6480154_2_P.fastq -S GCF_001887995.sam

777108 reads; of these:
  777108 (100.00%) were paired; of these:
    135263 (17.41%) aligned concordantly 0 times
    626930 (80.67%) aligned concordantly exactly 1 time
    14915 (1.92%) aligned concordantly >1 times
    ----
    135263 pairs aligned concordantly 0 times; of these:
      36305 (26.84%) aligned discordantly 1 time
    ----
    98958 pairs aligned 0 times concordantly or discordantly; of these:
      197916 mates make up the pairs; of these:
        193101 (97.57%) aligned 0 times
        3196 (1.61%) aligned exactly 1 time
        1619 (0.82%) aligned >1 times
87.58% overall alignment rate

real	6m1.824s
user	5m57.859s
sys	0m4.393s
```

### 3.5 Index and sort resulting SAM file

```sh
time samtools view -S -b GCF_001887995.sam | samtools sort - -o GCF_001887995.bam

real	0m59.706s
user	0m59.673s
sys	0m4.401s
```

### 3.6 Alignment statistics 

#### 3.6.1 Samtools Flagstat
```sh
samtools flagstat GCF_001887995.sam

1554216 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
1361115 + 0 mapped (87.58% : N/A)
1554216 + 0 paired in sequencing
777108 + 0 read1
777108 + 0 read2
1283690 + 0 properly paired (82.59% : N/A)
1357678 + 0 with itself and mate mapped
3437 + 0 singletons (0.22% : N/A)
202 + 0 with mate mapped to a different chr
120 + 0 with mate mapped to a different chr (mapQ>=5)

```

#### 3.6.1 Samtools Depth

```sh
samtools depth GCF_001887995.bam | awk '{sum+=$3;cnt++}END{print "Coverage: "sum/cnt" Sum: "sum}'
Coverage: 48.942 Sum: 242869272
```


